src/Lykke.Service.PaySettlement/Cqrs/CommandHandlers/CreateSettlementCommandHandler.cs

```
@@ -0,0 +1,208 @@
+                          $"{nameof(paymentAsset.LowVolumeAmount)} ({paymentAsset.LowVolumeAmount}).", new
+                {
+                    command.MerchantId,
+                    command.PaymentRequestId
+                });
+                return false;
+            }
+
+            return true;
+        }
+       
+        private async Task<string> GetMerchantClientIdAsync(CreateSettlementCommand command)
+        {
+            try
+            {
+                MerchantModel merchant = await _payMerchantClient.Api.GetByIdAsync(command.MerchantId);
+                return merchant.LwId;
+            }
+            catch (ClientApiException ex) when (ex.HttpStatusCode == HttpStatusCode.NotFound)
+            {
+                throw new SettlementException($"Merchant {command.MerchantId} is not found.", ex);
+            }
+        }
+
+        private bool ValidatePaymentRequest(IPaymentRequest paymentRequest)
+        {
+            if (string.IsNullOrEmpty(paymentRequest.MerchantClientId))
+            {
+                _log.Info("Skip payment request because merchant Lykke wallet is not set.", new
+                {
+                    paymentRequest.MerchantId,
+                    PaymentRequestId = paymentRequest.PaymentRequestId
+                });
+                return false;
+            }
+
+            return true;
+        }
+
+        private async Task TryAddPaymentRequestWithErrorAsync(IPaymentRequest paymentRequest, 
+            string errorMessage, IEventPublisher publisher)
+        {
+            if (paymentRequest == null)
+            {
+                return;
+            }
+
+            paymentRequest.Error = true;
+            paymentRequest.ErrorDescription = errorMessage;
+
+            try
+            {
+                await AddPaymentRequestAsync(paymentRequest, publisher);
+            }
+            catch (Exception)
+            {
```

Do you really not need to have handling code here?

---

This is "Try" method. It is used for store PaymentRequest in case of validating error.
It swallows exception like a TryParse methods.